vm/actor/src/builtin/miner/mod.rs

```diff
@@ -2020,113 +2820,100 @@ where
+            Serialized::serialize(OnMinerSectorsTerminateParamsRef {
+                epoch,
+                deal_ids: chunk,
+            })?,
+            TokenAmount::zero(),
+        )?;
     }
 
-    // let weights = sectors
-    //     .iter()
-    //     .map(|s| to_storage_weight_desc(sector_size, s))
-    //     .collect();
-
-    // TODO was refactored
-    // rt.send(
-    //     *STORAGE_POWER_ACTOR_ADDR,
-    //     PowerMethod::OnFaultEnd as u64,
-    //     Serialized::serialize(OnFaultEndParams { weights })?,
-    //     TokenAmount::zero(),
-    // )?;
     Ok(())
 }
 
-fn request_terminate_deals<BS, RT>(rt: &mut RT, deal_ids: Vec<DealID>) -> Result<(), ActorError>
+fn request_terminate_all_deals<BS, RT>(rt: &mut RT, state: &State) -> Result<(), ActorError>
 where
     BS: BlockStore,
     RT: Runtime<BS>,
 {
-    if deal_ids.is_empty() {
-        return Ok(());
-    }
+    let mut deal_ids = Vec::new();
 
-    rt.send(
-        *STORAGE_MARKET_ACTOR_ADDR,
-        MarketMethod::OnMinerSectorsTerminate as u64,
-        Serialized::serialize(OnMinerSectorsTerminateParams {
-            deal_ids,
-            // TODO this may not be correct (refactored)
-            epoch: rt.curr_epoch(),
-        })?,
-        TokenAmount::zero(),
-    )?;
-    Ok(())
+    state
+        .for_each_sector(rt.store(), |sector| {
+            deal_ids.extend_from_slice(&sector.deal_ids);
+            Ok(())
+        })
+        .map_err(|e| {
+            ActorError::new(
+                ExitCode::ErrIllegalState,
+                format!("failed to traverse sectors for termination: {:?}", e),
+            )
+        })?;
+
+    request_terminate_deals(rt, rt.curr_epoch(), deal_ids)
 }
-fn request_terminate_power<BS, RT>(
-    _rt: &mut RT,
-    _termination_type: SectorTermination,
-    _sector_size: SectorSize,
-    sectors: &[SectorOnChainInfo],
-) -> Result<(), ActorError>
+
+fn schedule_early_termination_work<BS, RT>(rt: &mut RT) -> Result<(), ActorError>
 where
     BS: BlockStore,
     RT: Runtime<BS>,
 {
-    if sectors.is_empty() {
-        return Ok(());
-    }
+    enroll_cron_event(
+        rt,
+        rt.curr_epoch() + 1,
+        CronEventPayload {
+            event_type: CRON_EVENT_PROCESS_EARLY_TERMINATIONS,
+        },
+    )
+}
 
-    // let weights = sectors
-    //     .iter()
-    //     .map(|s| to_storage_weight_desc(sector_size, s))
-    //     .collect();
-
-    // TODO was refactored
-    // rt.send(
-    //     *STORAGE_POWER_ACTOR_ADDR,
-    //     PowerMethod::OnSectorTerminate as u64,
-    //     Serialized::serialize(OnSectorTerminateParams {
-    //         termination_type,
-    //         weights,
-    //     })?,
-    //     TokenAmount::zero(),
-    // )?;
-    Ok(())
+fn have_pending_early_terminations<BS, RT>(_rt: &mut RT, state: &State) -> bool
+where
+    BS: BlockStore,
+    RT: Runtime<BS>,
+{
+    let no_early_terminations = state.early_terminations.is_empty();
+    !no_early_terminations
```

Comment:
```suggestion
    !state.early_terminations.is_empty()
```

Reply:
I know that this is insignificant either way but here I deliberately mirrored their variable name to make it easier to see that the logic matches (with the `empty` + `!` double negative)

Reply:
true!